From 02ad5cdde8284d006b1287c6a16f9c71a7531a8d Mon Sep 17 00:00:00 2001
From: Jean-Michel Bachot <jean-michel.bachot@intel.com>
Date: Fri, 22 Mar 2013 19:45:16 +0100
Subject: [PORT FROM MAIN] wifi: multi-vendor enabling in mainline

BZ: 60525

Mainline needs to support increasing number of devices.
Each device uses its own hardware mix.
This requires more and more scalability and flexibility.
This is especially true for connectivity where different
manufacturer might be involved.

In order to achieve that, these patches introduces a new way
of managing device configuration allowing us to easily decline
various hardware setup per device.

This patch make BCM wpa_supplicant fits the multi-vendor
configuration requirements.

Orig-Change-Id: I605d3a53246f7d193d5d74c4e6970f771d753336
Signed-off-by: Jean Trivelly <jean.trivelly@intel.com>
Reviewed-on: http://android.intel.com:8080/78739
Reviewed-by: Naoumenko, PaulX <paulx.naoumenko@intel.com>
Reviewed-by: Bachot, Jean-michel <jean-michel.bachot@intel.com>
Tested-by: Champciaux, Nicolas <nicolas.champciaux@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Jean-Michel Bachot <jean-michel.bachot@intel.com>

	modified:   Android.mk
	modified:   hostapd/Android.mk
	modified:   wpa_supplicant/Android.mk
	new file:   wpa_supplicant/AndroidLibWpaClient.mk

Change-Id: I4c38b287f5b84d677cc3cbf3cc776d675c0aff38
Signed-off-by: Jeremie GARCIA <jeremiex.garcia@intel.com>
---
 Android.mk                            |   13 +++++-----
 hostapd/Android.mk                    |    8 ++++--
 wpa_supplicant/Android.mk             |   33 +++++++++++++++-----------
 wpa_supplicant/AndroidLibWpaClient.mk |   41 +++++++++++++++++++++++++++++++++
 4 files changed, 71 insertions(+), 24 deletions(-)
 create mode 100644 wpa_supplicant/AndroidLibWpaClient.mk

diff --git a/Android.mk b/Android.mk
index 76afb77..860facc 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,13 +1,12 @@
-LOCAL_PATH:= $(call my-dir)
+LOCAL_DIR := $(call my-dir)
 
-ifndef WPA_SUPPLICANT_VERSION
-WPA_SUPPLICANT_VERSION := VER_0_8_X
-endif
-ifeq ($(WPA_SUPPLICANT_VERSION),VER_0_8_X)
+include  $(LOCAL_DIR)/wpa_supplicant/AndroidLibWpaClient.mk
+
+ifeq ($(strip $(BOARD_WLAN_DEVICE)),bcmdhd)
 # The order of the 2 Android.mks does matter!
 # TODO: Clean up the Android.mks, reset all the temporary variables at the
 # end of each Android.mk, so that one Android.mk doesn't depend on variables
 # set up in the other Android.mk.
-include $(LOCAL_PATH)/hostapd/Android.mk \
-        $(LOCAL_PATH)/wpa_supplicant/Android.mk
+include $(LOCAL_DIR)/hostapd/Android.mk \
+        $(LOCAL_DIR)/wpa_supplicant/Android.mk
 endif
diff --git a/hostapd/Android.mk b/hostapd/Android.mk
index 1c2e1f5..952650f 100644
--- a/hostapd/Android.mk
+++ b/hostapd/Android.mk
@@ -880,8 +880,9 @@ endif
 ########################
 
 include $(CLEAR_VARS)
-LOCAL_MODULE := hostapd_cli
-LOCAL_MODULE_TAGS := debug
+LOCAL_MODULE := hostapd_cli.bcm
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_STEM := hostapd_cli
 LOCAL_SHARED_LIBRARIES := libc libcutils liblog
 LOCAL_CFLAGS := $(L_CFLAGS)
 LOCAL_SRC_FILES := $(OBJS_c)
@@ -890,8 +891,9 @@ include $(BUILD_EXECUTABLE)
 
 ########################
 include $(CLEAR_VARS)
-LOCAL_MODULE := hostapd
+LOCAL_MODULE := hostapd.bcm
 LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_STEM := hostapd
 ifdef CONFIG_DRIVER_CUSTOM
 LOCAL_STATIC_LIBRARIES := libCustomWifi
 endif
diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 06beeaf..e57e236 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -1522,8 +1522,9 @@ endif
 ########################
 
 include $(CLEAR_VARS)
-LOCAL_MODULE := wpa_cli
-LOCAL_MODULE_TAGS := debug
+LOCAL_MODULE := wpa_cli.bcm
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_STEM := wpa_cli
 LOCAL_SHARED_LIBRARIES := libc libcutils liblog
 LOCAL_CFLAGS := $(L_CFLAGS)
 LOCAL_SRC_FILES := $(OBJS_c)
@@ -1531,8 +1532,11 @@ LOCAL_C_INCLUDES := $(INCLUDES)
 include $(BUILD_EXECUTABLE)
 
 ########################
+
 include $(CLEAR_VARS)
-LOCAL_MODULE := wpa_supplicant
+LOCAL_MODULE := wpa_supplicant.bcm
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_STEM := wpa_supplicant
 ifdef CONFIG_DRIVER_CUSTOM
 LOCAL_STATIC_LIBRARIES := libCustomWifi
 endif
@@ -1565,7 +1569,8 @@ LOCAL_COPY_HEADERS_TO := wpa_supplicant_8/src/common
 LOCAL_COPY_HEADERS :=\
     src/common/defs.h \
     src/common/ieee802_11_common.h \
-    src/common/ieee802_11_defs.h
+    src/common/ieee802_11_defs.h \
+    src/common/sae.h
 include $(BUILD_COPY_HEADERS)
 
 include $(CLEAR_VARS)
@@ -1639,13 +1644,13 @@ include $(BUILD_COPY_HEADERS)
 #include $(BUILD_PREBUILT)
 #
 ########################
-
-include $(CLEAR_VARS)
-LOCAL_MODULE = libwpa_client
-LOCAL_CFLAGS = $(L_CFLAGS)
-LOCAL_SRC_FILES = src/common/wpa_ctrl.c src/utils/os_$(CONFIG_OS).c
-LOCAL_C_INCLUDES = $(INCLUDES)
-LOCAL_SHARED_LIBRARIES := libcutils liblog
-LOCAL_COPY_HEADERS_TO := libwpa_client
-LOCAL_COPY_HEADERS := src/common/wpa_ctrl.h
-include $(BUILD_SHARED_LIBRARY)
+#
+# include $(CLEAR_VARS)
+# LOCAL_MODULE = libwpa_client
+# LOCAL_CFLAGS = $(L_CFLAGS)
+# LOCAL_SRC_FILES = src/common/wpa_ctrl.c src/utils/os_$(CONFIG_OS).c
+# LOCAL_C_INCLUDES = $(INCLUDES)
+# LOCAL_SHARED_LIBRARIES := libcutils liblog
+# LOCAL_COPY_HEADERS_TO := libwpa_client
+# LOCAL_COPY_HEADERS := src/common/wpa_ctrl.h
+# include $(BUILD_SHARED_LIBRARY)
diff --git a/wpa_supplicant/AndroidLibWpaClient.mk b/wpa_supplicant/AndroidLibWpaClient.mk
new file mode 100644
index 0000000..e81d61d
--- /dev/null
+++ b/wpa_supplicant/AndroidLibWpaClient.mk
@@ -0,0 +1,41 @@
+LOCAL_PATH := $(call my-dir)
+
+#######################
+LIBWPA_CLIENT_CONFIG_OS = unix
+
+include $(CLEAR_VARS)
+LOCAL_MODULE := libwpa_client
+LOCAL_CFLAGS = \
+   -DCONFIG_CTRL_IFACE \
+   -DCONFIG_CTRL_IFACE_UNIX \
+   -DANDROID \
+   -DCONFIG_BGSCAN_LEARN \
+   -DCONFIG_CTRL_IFACE_CLIENT_DIR=\"/data/misc/wifi/sockets\" \
+   -DCONFIG_CTRL_IFACE_DIR=\"/data/system/wpa_supplicant\" \
+   -DCONFIG_CTRL_IFACE_DBUS_NEW
+LOCAL_SRC_FILES = src/common/wpa_ctrl.c src/utils/os_$(LIBWPA_CLIENT_CONFIG_OS).c
+LOCAL_C_INCLUDES = \
+   $(LOCAL_PATH) \
+   $(LOCAL_PATH)/src \
+   $(LOCAL_PATH)/src/common \
+   $(LOCAL_PATH)/src/drivers \
+   $(LOCAL_PATH)/src/eap_common \
+   $(LOCAL_PATH)/src/eapol_supp \
+   $(LOCAL_PATH)/src/eap_peer \
+   $(LOCAL_PATH)/src/eap_server \
+   $(LOCAL_PATH)/src/hlr_auc_gw \
+   $(LOCAL_PATH)/src/l2_packet \
+   $(LOCAL_PATH)/src/radius \
+   $(LOCAL_PATH)/src/rsn_supp \
+   $(LOCAL_PATH)/src/tls \
+   $(LOCAL_PATH)/src/utils \
+   $(LOCAL_PATH)/src/wps \
+   external/openssl/include \
+   system/security/keystore \
+   external/libnl-headers
+LOCAL_MODULE_TAGS := optional
+LOCAL_SHARED_LIBRARIES := libcutils
+LOCAL_COPY_HEADERS_TO := libwpa_client
+LOCAL_COPY_HEADERS := src/common/wpa_ctrl.h
+include $(BUILD_SHARED_LIBRARY)
+
-- 
1.7.4.1

